net/tstun/table/table.go

```diff
@@ -0,0 +1,58 @@
+// Copyright (c) Tailscale Inc & AUTHORS
+// SPDX-License-Identifier: BSD-3-Clause
+
+// Package table provides a Routing Table implementation which allows
+// looking up the peer that should be used to route a given IP address.
+package table
+
+import (
+	"net/netip"
+
+	"tailscale.com/tempfork/device"
+	"tailscale.com/types/key"
+	"tailscale.com/util/mak"
+)
+
+// RoutingTable provides a mapping from IP addresses to peers identified by
+// their public node key. It is used to find the peer that should be used to
+// route a given IP address.
+//
+// It is not safe for concurrent use.
+type RoutingTable struct {
+	peers      map[key.NodePublic]*device.Peer
+	allowedIPs device.AllowedIPs
```

Comment:
Better name plz, took a few takes to figure out what this was meant to be

Reply:
went with prefixTrie and added docs